avocado/core/loader.py

```diff
@@ -235,6 +235,10 @@ def load_test(self, test_factory):
         :return: an instance of :class:`avocado.core.test.Test`.
         """
         test_class, test_parameters = test_factory
+        # discard tags, as they are *not* intented to be parameters
+        # for the test, but used previously during filtering
+        if 'tags' in test_parameters:
+            del(test_parameters['tags'])
```

Comment:
Passing tags via that way is identical to how we deal with `methodName` for instrumented tests.  It's not part of a documented API (ad-hoc indeed), but it's in line with the current approach.

I'll all in favor of giving the loader API a good look, including documenting the basic interfaces, making them "new style" plugins, etc.

Do you think we should work on those before adding this feature?  I honestly (and obviously since I'm proposing this), don't. 😄